client/my-sites/upgrades/domain-management/dns/dns-add-new.jsx

```diff
@@ -33,7 +33,8 @@ const DnsAddNew = React.createClass( {
 	getInitialState() {
 		return {
 			show: false,
-			fields: null
+			fields: null,
+			submittingForm: false,
```

Comment:
What about moving this state [in the store](https://github.com/Automattic/wp-calypso/blob/master/client/lib/domains/site-redirect/reducer.js#L24) instead? That's what we've been doing for some time now and it works [pretty well](https://github.com/Automattic/wp-calypso/blob/master/client/my-sites/upgrades/domain-management/site-redirect/index.jsx#L111). It helps to centralize state management and to reduce the complexity of components.